### PR TITLE
Print extra info for util/readdmtcp.sh

### PR DESCRIPTION
**Pull request by gc00**
This improves on util/readdmtcp.sh by having `mtcp_restart --simulate` also print out most of the information in `mtcpHdr`, such as vdso, vvar, postRestart entry point, etc.

### CODE REVIEW

File: src/mtcp/mtcp_restart.c [hunk 1]
```diff
@@ -207,7 +207,7 @@ MTCP_PRINTF("Attach for debugging.");
   while (argc > 0) {
     // Flags for standalone debugging
     if (argc == 1) {
-      MTCP_PRINTF("Considering '%s' as a ckpt image.\n", argv[0]);
+      mtcp_printf("Considering '%s' as a ckpt image.\n", argv[0]);
```
Comment (karya0) [inline src/mtcp/mtcp_restart.c]: Why this change? We decided to use MTCP_PRINTF and DPRINTF exclusively, unless we are printing some raw data where we don't want to print line/function information with every line.

File: src/mtcp/mtcp_restart.c [hunk 2]
```diff
@@ -463,11 +463,25 @@ static void restart_slow_path()
...
-static void mtcp_simulateread(int fd)
+static void mtcp_simulateread(int fd, MtcpHeader *mtcpHdr)
 {
   int mtcp_sys_errno;
+
+  // Print miscellaneous information:
+  char buf[MTCP_SIGNATURE_LEN+1];
+  mtcp_memcpy(buf, mtcpHdr->signature, MTCP_SIGNATURE_LEN);
+  buf[MTCP_SIGNATURE_LEN] = '\0';
+  mtcp_printf("\nMTCP: %s", buf);
```
Comment (karya0) [inline src/mtcp/mtcp_restart.c]: Here it is okay to use `mtcp_printf` because we are printing a lot of raw data. Can you also add a MTCP_PRINTF at the top of this function that would act as a header for this raw data?

File: src/mtcp/mtcp_restart.c [hunk 3]
```diff
@@ -463,11 +463,25 @@ static void restart_slow_path()
...
+              mtcpHdr->restore_addr + mtcpHdr->restore_size);
+  mtcp_printf("**** DMTCP entry point (ThreadList::postRestart()): %p\n",
+              mtcpHdr->post_restart);
+  mtcp_printf("**** brk (sbrk(0)): %p\n", mtcpHdr->saved_brk);
+  mtcp_printf("**** vdso: %p-%p\n", mtcpHdr->vdsoStart, mtcpHdr->vdsoEnd);
+  mtcp_printf("**** vvar: %p-%p\n", mtcpHdr->vvarStart, mtcpHdr->vvarEnd);
+
   Area area;
-  MTCP_PRINTF("Listing ckpt image area:\n");
+  mtcp_printf("\n**** Listing ckpt image area:\n");
```
Comment (karya0) [inline src/mtcp/mtcp_restart.c]: This should be promoted to `MTCP_PRINTF` as well since it's a header.

File: src/mtcp/mtcp_restart.c (hunk 2 above)
Comment (gc00) [inline src/mtcp/mtcp_restart.c]: The primary use of this function is to print to stdout when called by readdmtcp.sh.  The routine MTCP_PRINTF() will print the file, function, line number, etc., followed by an indented message on the next line.  I don't think people calling readdmtcp.sh would want to see what function and line number is delivering the information to them.  (For example, readelf doesn't do that.)

File: src/mtcp/mtcp_restart.c (hunk 3 above)
Comment (gc00) [inline src/mtcp/mtcp_restart.c]: Same comment as before.  readelf.sh doesn't report the internal function and line number delivering the information to us.  util/readdmtcp.sh shouldn't be doing that either.

File: src/mtcp/mtcp_restart.c (hunk 1 above)
Comment (karya0) [inline src/mtcp/mtcp_restart.c]: Could you also add a comment here that states the fact that this condition is true only for readmtcp and that's why it's okay to use mtcp_printf here. Alternatively, could we just remove this line from here and move it in `mtcp_simulateread`?

File: src/mtcp/mtcp_restart.c [hunk 4]
```diff
@@ -207,7 +207,8 @@ MTCP_PRINTF("Attach for debugging.");
   while (argc > 0) {
     // Flags for standalone debugging
     if (argc == 1) {
-      MTCP_PRINTF("Considering '%s' as a ckpt image.\n", argv[0]);
+      // We would use MTCP_PRINTF, but it's also for output of util/readdmtcp.sh
```
Comment (gc00) [inline src/mtcp/mtcp_restart.c]: This comment was added, based on the review.

**Comment by gc00** [comment]
I've added the comment on line 10 as proposed.  Let me know if it's ready to go.

**Comment by karya0** [comment]
It's ready to go. If you get a chance, can you add a short comment at the top of `mtcp_simulateread` explaining the mtcp_printf vs MTCP_PRINTF choice.

**Comment by gc00** [comment]
Done.  :-)
